discretize/TensorMesh.py

```diff
@@ -598,56 +598,65 @@ class TensorMesh(
     def __init__(self, h=None, x0=None, **kwargs):
         BaseTensorMesh.__init__(self, h=h, x0=x0, **kwargs)
 
-    def __str__(self):
-        outStr = '  ---- {0:d}-D TensorMesh ----  '.format(self.dim)
-        def printH(hx, outStr=''):
-            i = -1
-            while True:
-                i = i + 1
-                if i > hx.size:
-                    break
-                elif i == hx.size:
-                    break
-                h = hx[i]
-                n = 1
-                for j in range(i+1, hx.size):
-                    if hx[j] == h:
-                        n = n + 1
-                        i = i + 1
-                    else:
-                        break
-
-                if n == 1:
-                    outStr += ' {0:.2f}, '.format(h)
-                else:
-                    outStr += ' {0:d}*{1:.2f}, '.format(n, h)
-
-            return outStr[:-1]
-
-        if self.dim == 1:
-            outStr += '\n   x0: {0:.2f}'.format(self.x0[0])
-            outStr += '\n  nCx: {0:d}'.format(self.nCx)
-            outStr += printH(self.hx, outStr='\n   hx:')
-            pass
-        elif self.dim == 2:
-            outStr += '\n   x0: {0:.2f}'.format(self.x0[0])
-            outStr += '\n   y0: {0:.2f}'.format(self.x0[1])
-            outStr += '\n  nCx: {0:d}'.format(self.nCx)
-            outStr += '\n  nCy: {0:d}'.format(self.nCy)
-            outStr += printH(self.hx, outStr='\n   hx:')
-            outStr += printH(self.hy, outStr='\n   hy:')
-        elif self.dim == 3:
-            outStr += '\n   x0: {0:.2f}'.format(self.x0[0])
-            outStr += '\n   y0: {0:.2f}'.format(self.x0[1])
-            outStr += '\n   z0: {0:.2f}'.format(self.x0[2])
-            outStr += '\n  nCx: {0:d}'.format(self.nCx)
-            outStr += '\n  nCy: {0:d}'.format(self.nCy)
-            outStr += '\n  nCz: {0:d}'.format(self.nCz)
-            outStr += printH(self.hx, outStr='\n   hx:')
-            outStr += printH(self.hy, outStr='\n   hy:')
-            outStr += printH(self.hz, outStr='\n   hz:')
-
-        return outStr
+    def _get_attrs(self):
+        """An internal helper for the representation methods"""
+        attrs = []
+        dims = ['x', 'y', 'z']
```

Comment:
this could be a property on the mesh - it would make it easier to inherit (then on the cyl mesh for example, all of this code could be re-used, but `dims = ['r', 'theta', 'z']`